modules/apps/transfer/module.go

```diff
@@ -144,17 +140,18 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 
 // InitGenesis performs genesis initialization for the ibc-transfer module. It returns
 // no validator updates.
-func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.RawMessage) {
+func (am AppModule) InitGenesis(ctx context.Context, data json.RawMessage) error {
 	var genesisState types.GenesisState
-	cdc.MustUnmarshalJSON(data, &genesisState)
+	am.cdc.MustUnmarshalJSON(data, &genesisState)
```

Comment:
switch to non panic-able `UnmarshalJSON` and return err?

Reply:
can be done, not strictly required but now we have genesis error returns I think this could be implemented across the board (many genesis funcs still panic iirc)

Reply:
#7698